plutus-core/plutus-ir/test/PlutusIR/Transform/StrictLetRec/Tests.hs

```diff
@@ -0,0 +1,53 @@
+{-# LANGUAGE BlockArguments    #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module PlutusIR.Transform.StrictLetRec.Tests where
+
+import PlutusPrelude
+
+import Control.Monad.Except (runExcept)
+import Control.Monad.Reader (runReaderT)
+import PlutusCore.Default (someValue)
+import PlutusCore.MkPlc (constant)
+import PlutusCore.Quote (runQuoteT)
+import PlutusIR.Compiler.Let (LetKind (RecTerms), compileLetsPassSC)
+import PlutusIR.Compiler.Provenance (noProvenance)
+import PlutusIR.Parser (pTerm)
+import PlutusIR.Pass.Test (runTestPass)
+import PlutusIR.Test (goldenPirM)
+import PlutusIR.Transform.StrictLetRec.Tests.Lib (defaultCompilationCtx,
+                                                  evalPirProgramWithTracesOrFail, pirTermAsProgram,
+                                                  pirTermFromFile)
+import System.FilePath.Posix (joinPath, (</>))
+import Test.Tasty (TestTree)
+import Test.Tasty.Extras (runTestNestedIn, testNested)
+import Test.Tasty.HUnit (testCase, (@?=))
+import UntypedPlutusCore.Evaluation.Machine.Cek (EvaluationResult (..))
+
+path :: [FilePath]
+path = ["plutus-ir", "test", "PlutusIR", "Transform"]
+
+test_letRec :: TestTree
+test_letRec = runTestNestedIn path do
+  testNested
+    "StrictLetRec"
+    [ let
+        runCompilationM m = either (fail . show) pure do
+          ctx <- defaultCompilationCtx
+          runExcept . runQuoteT $ runReaderT m ctx
+       in
+        goldenPirM
+          (runCompilationM . runTestPass (`compileLetsPassSC` RecTerms))
+          (const noProvenance <<$>> pTerm)
+          "strictLetRec"
+    , pure $ testCase "traces" do
+        (result, traces) <- do
+          pirTerm <- pirTermFromFile (joinPath path </> "StrictLetRec" </> "strictLetRec")
+          evalPirProgramWithTracesOrFail (pirTermAsProgram (void pirTerm))
+        case result of
+          EvaluationFailure ->
+            fail $ "Evaluation failed, available traces: " <> show traces
+          EvaluationSuccess term -> do
+            term @?= constant () (someValue (1 :: Integer))
```

Comment:
@Unisay just FYI, you can write `fromValue (1 :: Integer)` instead of `constant () (someValue (1 :: Integer)`. It's a bit confusing, because we have a `HasConstant` class that contains

```haskell
    -- | Wrap a Haskell value as a @term@.
    fromConstant :: Some (ValueOf (UniOf term)) -> term
```

which is a lot like `TermLike` containing

```haskell
    constant :: ann -> Some (ValueOf uni) -> term ann
```

but doesn't take an annotation. `fromValue` is defined in terms of the former and we use it during evaluation unlike anything from `TermLike`. And in general, evaluation stuff is separate (because it needs to be ultra-optimized), so it doesn't have great discoverability.

Reply:
Thanks, will use `fromValue` next time!